### PR TITLE
Feat/tcp support

### PR DESCRIPTION
**Pull request by mjlbach**
This is more of a proof of concept to show how it should be done. There's a cleaner way (that should support arbitrary streams for distant.nvim and co), I just wanted to get this off the ground by doing the preliminary refactoring. Right now it's hardcoded to work with godot (the only TCP server that I know off the top of my head)

### CODE REVIEW

File: runtime/lua/vim/lsp.lua [hunk 1]
```diff
@@ -798,11 +800,27 @@ function lsp.start_client(config)
     end
   end
 
+  local handles
+  -- Start the language server
+  if config.cmd then
```
Comment (mjlbach) [inline runtime/lua/vim/lsp.lua:805]: this should probably switch on some `mode` key to config, you can start a server *and* connect to it via TCP

Comment (AusCyberman) [reply runtime/lua/vim/lsp.lua:805]: > this should probably switch on some `mode` key to config, you can start a server _and_ connect to it via TCP

like there should also be a named pipe/ unix socket option as well, as some servers (`powershell_es` for example) use named pipes to communicate

Comment (mjlbach) [reply runtime/lua/vim/lsp.lua:805]: Sure, named pipe support could be added but is out of scope of this PR which adds TCP support.

**Comment by AusCyberman** [comment]
it could be a variable called `channel` and be either `stdio` or `{ip = "127.0.0.1"}`.
it also could be a thing called ip address, and port like
```lua
if config.port then
local ip_address = config.ip or "127.0.0.1
--- etc
```

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Pull request closed without merging**